pkg/devfile/adapters/docker/utils/utils.go

```diff
@@ -135,3 +179,106 @@ func containerHasPort(devfilePort nat.Port, exposedPorts nat.PortSet) bool {
 	}
 	return false
 }
+
+// UpdateComponentWithSupervisord updates the devfile component's
+// 1. command and args with supervisord, if absent
+// 2. env with ODO_COMMAND_RUN and ODO_COMMAND_RUN_WORKING_DIR, if absent
+func UpdateComponentWithSupervisord(comp *common.DevfileComponent, runCommand common.DevfileCommand, supervisordVolumeName string, hostConfig *container.HostConfig) {
```

Comment:
I don't think we check once if `comp` is nil? Could have a race condition here in the future.

Reply:
There are no tests for this...

Reply:
comp is first received during odo push here https://github.com/openshift/odo/blob/master/pkg/devfile/adapters/common/utils.go#L66; we use this `comp` to already pull images and start containers and while we loop thru the comps, we call this function